dataRetriever/requestHandlers/requestHandler.go

```diff
@@ -119,6 +97,8 @@ func (rrh *resolverRequestHandler) requestByHashes(destShardID uint32, hashes []
 		return
 	}
 
+	rrh.whiteList.Add(hashes)
```

Comment:
Should't the hashes be added to the whitelist only if the request was successful? If yes, move this to line 124.

Reply:
done.